clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -1747,11 +1753,17 @@ private String updatePushJobDetailsWithJobDetails(DataWriterTaskTracker dataWrit
     final long recordTooLargeFailureCount = dataWriterTaskTracker.getRecordTooLargeFailureCount();
     if (recordTooLargeFailureCount > 0) {
       updatePushJobDetailsWithCheckpoint(PushJobCheckpoints.RECORD_TOO_LARGE_FAILED);
-      String errorMessage = String.format(
-          "Input data has at least %d records that exceed the maximum record limit of %s",
+
+      // maxSizeForUserPayloadPerMessageInBytes < 1MB (Kafka record limit); errors appear when chunking is not enabled
+      // maxRecordSizeBytes can be much larger and should take effect only once chunking is enabled
+      final int recordSizeLimit = (pushJobDetails.chunkingEnabled)
```

Comment:
That's a good catch. Maybe it would be better to set the default limit in `VenicePushJob` as opposed to [centrally in `VeniceWriter`](https://github.com/linkedin/venice/pull/1052#discussion_r1672703648) then.

Reply:
I think the default limit should be controlled by via a backend config in Controller, and the config can be in cluster level.
So when Controller returns the store response, it can backfill the record size limit by the config if the value is `-1`.

Reply:
Will do.